reference/constraints/Timezone.rst

```diff
@@ -85,8 +86,28 @@ string meant to contain a timezone identifier (e.g. ``America/New_York``):
 Options
 -------
 
+countryCode
+~~~~~~~~~~~
+
+**type**: ``string`` **default**: ``null``
+
+If the ``zone`` option is set to ``\DateTimeZone::PER_COUNTRY``, this option
+restricts the valid timezone identifiers to the ones that belong to the given
+country.
+
+The value of this option must be a valid `ISO 3166-1 alpha-2`_ country code
+(e.g. ``CN`` for China).
+
 .. include:: /reference/constraints/_groups-option.rst.inc
 
+intlCompatible
+~~~~~~~~~~~~~~
+
+**type**: ``boolean`` **default**: ``false``
+
+If ``true``, this constraint also considers valid any of the
+:ref:`ICU timezones <component-inlt-timezones>` provided by the Intl component.
```

Comment:
this happens out-of-the-box actually, depending if Intl is available yes/no

given the Intl component is a compiled version of ICU, the `intlCompatible` option ensures the timezone is valid using the current runtime version of ICU (ext-intl), i.e. when using `IntlTimeZone::create`